src/tlo/methods/healthsystem.py

```diff
@@ -2445,17 +2462,58 @@ def process_events_mode_2(self, hold_over: List[HSIEventQueueItem]) -> None:
                         # based on queue information, and we assume no squeeze ever takes place.
                         squeeze_factor = 0.
 
-                        # Check if any of the officers required have run out.
+                        # Check if any of the officers required have run out. If including task-shifting,
+                        # this will involve checking if alternative capabilities are available for officers
+                        # that are no longer available.
                         out_of_resources = False
+                        
+                        # This dictionary stores all task-shifting officers considered for this appointment
+                        task_shifting_adopted = {}
```

Comment:
this dict isn't logged or used. 

To allow logging, we _could_ save a flag onto the HSI Event itself (e.g. `.global_task_shifting_used: bool`) and then use this property when do the logging (summary and detail).

Reply:
It is adopted, and it must be used to know which officers were assigned what tasks (line 2506)

Reply:
I see it being written to but not read from (?)

Reply:
Lines 2593 to 2596